eclipse-collections/src/main/java/org/eclipse/collections/impl/bag/immutable/ImmutableBagFactoryImpl.java

```diff
@@ -105,4 +108,55 @@
         }
         return this.of((T[]) Iterate.toArray(items));
     }
+
+    @Override
+    public <T> ImmutableBag<T> withOccurrences(T element, int occurrence)
+    {
+        MutableBag<T> bag = new HashBag<>(occurrence);
+        bag.addOccurrences(element, occurrence);
+
+        return bag.toImmutable();
+    }
+
+    @Override
+    public <T> ImmutableBag<T> withOccurrences(T element1, int occurrence1, T element2, int occurrence2)
+    {
+        MutableBag<T> bag = new HashBag<>(occurrence1 + occurrence2);
+        bag.addOccurrences(element1, occurrence1);
+        bag.addOccurrences(element2, occurrence2);
+        return bag.toImmutable();
+    }
+
+    @Override
+    public <T> ImmutableBag<T> withOccurrences(T element1, int occurrence1, T element2, int occurrence2, T element3, int occurrence3)
+    {
+        MutableBag<T> bag = new HashBag<>(occurrence1 + occurrence2 + occurrence3);
+        bag.addOccurrences(element1, occurrence1);
+        bag.addOccurrences(element2, occurrence2);
+        bag.addOccurrences(element3, occurrence3);
+        return bag.toImmutable();
+    }
+
+    @Override
+    public <T> ImmutableBag<T> withOccurrences(T element1, int occurrence1, T element2, int occurrence2, T element3, int occurrence3, T element4, int occurrence4)
+    {
+        MutableBag<T> bag = new HashBag<>(occurrence1 + occurrence2 + occurrence3 + occurrence4);
+        bag.addOccurrences(element1, occurrence1);
+        bag.addOccurrences(element2, occurrence2);
+        bag.addOccurrences(element3, occurrence3);
+        bag.addOccurrences(element4, occurrence4);
+        return bag.toImmutable();
+    }
+
+    @Override
+    public <T> ImmutableBag<T> withOccurrences(ObjectIntPair<T>... elementsWithOccurrences)
+    {
+        MutableBag<T> bag = new HashBag<>();
```

Comment:
This is the only location where pre-sizing may have a benefit. The size would be the array length.